tests/test_leftovers.py

```diff
@@ -0,0 +1,294 @@
+        "size": 123,
+        "hashes": {"md5": "8b9942cf415384b27cadf1f4d2d682e5"},
+        "release": "private",
+        "keeper_authority": "CRI",
+        "host_authorities": ["PDC"],
+    }
+    ark = "ark:/31807/TEST-abc"
+    res = client.put("/alias/" + ark, json=data, headers=user)
+    assert res.status_code == 200
+    rec = res.json
+    assert rec["name"] == ark
+    # assert there is only one entry in the alias index
+    assert len(client.get("/alias/").json["aliases"]) == 1
+    # assert that the returned alias by size is the same as the posted
+    res = client.get("/alias/?size=" + str(data["size"]))
+    rec = res.json
+    assert rec["aliases"][0] == ark
+
+
+def test_index_stats(client, user):
+    # populate the index with three different size records
+    data1 = get_doc()
+    res = client.post("/index/", json=data1, headers=user)
+    assert res.status_code == 200
+    data2 = get_doc()
+    data2["size"] = 77
+    res = client.post("/index/", json=data2, headers=user)
+    assert res.status_code == 200
+    data3 = get_doc()
+    data3["size"] = 300
+    res = client.post("/index/", json=data3, headers=user)
+    assert res.status_code == 200
+    data_size = data1["size"] + data2["size"] + data3["size"]
+    index_stats = client.get("/_stats/").json
+    file_number = index_stats["fileCount"]
+    index_file_size = index_stats["totalFileSize"]
+    # test that the stat file number and size is consistent with post
+    assert file_number == 3
+    assert index_file_size == data_size
+
+
+def test_status_check(client, user):
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/_status/")
+    assert res.status_code == 200
+
+
+def test_version_check(client, user):
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/_version")
+    assert res.status_code == 200
+
+
+def test_index_list_by_size(client, user):
+    # post two records of different size
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["size"] = 100
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?size=" + str(data["size"]))
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper size
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["size"] == 100
+
+
+def test_index_list_by_file_name(client, user):
+    # post two records of different name
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["file_name"] = "test_file"
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?file_name=" + data["file_name"])
```

Comment:
your test case is `1 record with a file name + 1 record with no file name; query with the file name`. but this doesn't test that records that _have_ file names that are different from the one you query are not returned

could you add a 3rd record with a different file name?